src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -157,6 +162,13 @@ public NadelExecutionStrategy(List<Service> services,
 
                 ExecutionContext newExecutionContext = buildServiceVariableOverrides(executionContext, queryTransform.getVariableValues());
 
+                String topLevelFieldId = esi.getFieldDefinition().getDefinition().getAdditionalData().get("id");
+                Optional<TransformationMetadata.NormalizedFieldAndError> maybeTopLevelFieldError = queryTransform.getRemovedFieldMap().getRemovedFieldById(topLevelFieldId);
+                boolean topLevelFieldExecutionShouldBeSkipped = maybeTopLevelFieldError.isPresent();
+                if (topLevelFieldExecutionShouldBeSkipped) {
```

Comment:
Move the isPresent check into a util function inside `TransformationMetadata`

That way we can do

```java
var removedFields = queryTransform.getRemovedFieldMap();
if (removedFields.isFieldRemoved(topLevelFieldId)) {
    …
}
```

It's a small change but when your codebase is huge and complex the readability gains are worth it.

Reply:
Not sure about this one. Are you saying the method `getRemovedFieldById(topLevelFieldId)` should stay or be gone? If it stays, does it still return an Optional?

Reply:
I'm saying the checking logic here should be simplified.

There should be a method called `isFieldRemoved` that encapsulates the logic here

```java
public boolean isFieldRemoved(String fieldId) {
    return getRemovedFieldById(fieldId).isPresent();
}
```

So you call

```java
var removedFields = queryTransform.getRemovedFieldMap();
if (removedFields.isFieldRemoved(topLevelFieldId)) {
    …
}
```

It just reads a lot better than getting the removed field only to call is present on it

Reply:
yeah but if we do what you are suggesting we will need to call `isFieldRemoved` and then we will need to get the removed field data (the error) anyway. This is why I am asking
> Are you saying the method getRemovedFieldById(topLevelFieldId) should stay or be gone?
I don't think it's providing much value to do what you are suggesting if we need to check `isFieldRemoved` first and then get the field.

Reply:
Ah shit I completely missed the part where you use the variable afterwards. Yea ok my bad. I thought we just got the error just to check if it was present.